src/abstractdataframe/show.jl

```diff
@@ -113,7 +119,11 @@ function compacttype(T::Type, maxwidth::Int=8, initial::Bool=true)
 
     maxwidth -= 1 # we will add "…" at the end
 
-    if T <: CategoricalValue
+    # assume that if module of T is CategoricalArrays its name is CategoricalValue
+    # the T is an instance of CategoricalValue from CategoricalArrays.jl
+    if hasproperty(T, :name) &&
+        nameof(T.name.module) === :CategoricalArrays &&
+        T.name.name === :CategoricalValue
```

Comment:
Do we not care if the `T.parameters[1] <: AbstractString` here?

Reply:
bacause we are replacing the behaviour for `CategoricalValue{<:AbstractString}` defined earlier in line https://github.com/JuliaData/DataFrames.jl/pull/2427/files#diff-38a6c4f0cb7743ccc6d4e8470ebdbd11L69. I just want to ensure I am matching functionality 1-2-1 exactly.